src/device/GPU_Device.cc

```diff
@@ -26,143 +27,75 @@ namespace rtt_device {
+                             device_properties.maxThreadsDim[2]});
+      m_maxgridsize.push_back(std::array<int, 3>{
+          device_properties.maxGridSize[0], device_properties.maxGridSize[1],
+          device_properties.maxGridSize[2]});
+      m_sharedmemperblock.push_back(device_properties.sharedMemPerBlock);
 
       err =
-          cuDeviceGetAttribute(&tmp1, CU_DEVICE_ATTRIBUTE_WARP_SIZE, cuDevice);
+          cudaDeviceGetAttribute(&tmp, cudaDevAttrTotalConstantMemory, device);
       checkForCudaError(err);
-      m_simdwidth.push_back(tmp1);
+      m_totalconstantmemory.push_back(tmp);
 
-      err =
-          cuDeviceGetAttribute(&tmp1, CU_DEVICE_ATTRIBUTE_MAX_PITCH, cuDevice);
-      checkForCudaError(err);
-      m_mempitch.push_back(tmp1);
-
-      err = cuDeviceGetAttribute(
-          &tmp1, CU_DEVICE_ATTRIBUTE_MAX_REGISTERS_PER_BLOCK, cuDevice);
-      checkForCudaError(err);
-      m_regsperblock.push_back(tmp1);
+      m_simdwidth.push_back(device_properties.warpSize);
+      m_mempitch.push_back(device_properties.memPitch);
 
       err =
-          cuDeviceGetAttribute(&tmp1, CU_DEVICE_ATTRIBUTE_CLOCK_RATE, cuDevice);
+          cudaDeviceGetAttribute(&tmp, cudaDevAttrMaxRegistersPerBlock, device);
       checkForCudaError(err);
-      m_clockrate.push_back(tmp1);
+      m_regsperblock.push_back(tmp);
 
-      err = cuDeviceGetAttribute(&tmp1, CU_DEVICE_ATTRIBUTE_TEXTURE_ALIGNMENT,
-                                 cuDevice);
+      err = cudaDeviceGetAttribute(&tmp, cudaDevAttrClockRate, device);
       checkForCudaError(err);
-      m_texturealign.push_back(tmp1);
-    }
-
-    // CUdevprop_st properties;
-    // err = cuDeviceGetProperties(&properties, cuDevice);
-    // checkForCudaError(err);
-    // deviceProperties.push_back(properties);
-  }
+      m_clockrate.push_back(tmp);
 
-  // Save the handle and context for each device
-  device_handle.resize(2);
-  context.resize(2);
-  for (int device = 0; device < deviceCount; device++) {
-    // Only initialize if compute capability >= 2.0
-    if (computeCapability[device][0] >= 2) {
-      // Save the handle for each device
-      err = cuDeviceGet(&device_handle[device], device);
-      checkForCudaError(err);
-
-      // Save the handle for each context
-      err = cuCtxCreate(&context[device], device, device_handle[device]);
+      err = cudaDeviceGetAttribute(&tmp, cudaDevAttrTextureAlignment, device);
       checkForCudaError(err);
+      m_texturealign.push_back(tmp);
     }
   }
 }
 
-/*!
- * \brief destructor
- *
- * Free the device context and unload any modules.
- */
-GPU_Device::~GPU_Device() {
-  // Free reserved contexts:
-  for (int device = 0; device < deviceCount; device++) {
-    // Only initialize if compute capability >= 2.0
-    if (computeCapability[device][0] >= 2) {
-      cudaError_enum err = cuCtxDestroy(context[device]);
-      checkForCudaError(err);
-    }
-  }
-}
+GPU_Device::~GPU_Device() {}
```

Comment:
So cuda no longer requires initialization and destruction to use a device?  That is a real improvement!